### PR TITLE
Fix uint32_t printed as unsigned int

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
This is ok in practice since we don't support 16-bit platforms, but it makes `arm-none-eabi-gcc-10 -mthumb -Wformat` complain.

Fixes `build_arm_none_eabi_gcc_no_64bit_multiplication` on my machine (arm-none-eabi-gcc 10.3). The CI doesn't complain because we're using an older compiler version.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required (just warnings that nobody's complained about)
- [x] **3.6 backport** at https://github.com/Mbed-TLS/mbedtls/pull/9231
- [x] **2.28 backport** no (code added later)
- [x] **tests** not practical; will be tested when we get around to upgrading the compiler on the CI

### CODE REVIEW

File: library/ssl_tls13_generic.c [hunk 1]
```diff
@@ -1484,9 +1484,12 @@ int mbedtls_ssl_tls13_check_early_data_len(mbedtls_ssl_context *ssl,
          ssl->total_early_data_size)) {
 
         MBEDTLS_SSL_DEBUG_MSG(
-            2, ("EarlyData: Too much early data received, %u + %" MBEDTLS_PRINTF_SIZET " > %u",
-                ssl->total_early_data_size, early_data_len,
-                ssl->session_negotiate->max_early_data_size));
+            2, ("EarlyData: Too much early data received, "
+                "%" MBEDTLS_PRINTF_SIZET " + %" MBEDTLS_PRINTF_SIZET
+                " > %" MBEDTLS_PRINTF_SIZET,
+                (size_t) ssl->total_early_data_size,
```
Comment (tom-cosgrove-arm) [inline library/ssl_tls13_generic.c:1490]: If we're going to cast a `uint32_t`, why not cast to `long` so we can just use `%lu` - smaller change and the resulting code is easier to read (I mean, given we have an existing `MBEDTLS_PRINTF_SIZET` it's not great at the moment, of course)

Comment (gilles-peskine-arm) [reply library/ssl_tls13_generic.c]: I don't think mixing types makes the instruction as a whole more readable, but it works either way.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by tom-cosgrove-arm** [comment]
Is there a problem with the CI? DCO status is normally reported quickly

**Comment by gilles-peskine-arm** [comment]
DCO occasionally gets lost on the way. I've force-pushed a new identical commit to kick it again.

**Comment by gilles-peskine-arm** [comment]
Not helping and https://github.com/Mbed-TLS/mbedtls/pull/9189's DCO also hasn't reported back. The DCO job is down.

**Review by paul-elliott-arm — APPROVED** [review]
LGTM, DCO is good by manual inspection

**Comment by gilles-peskine-arm** [comment]
I force-pushed a new head with only a commit date change to re-run the CI (DCO service outage, and unrelated intermittent failure on the internal CI that's now fixed).

**Pull request merged**